internal/metrics/gitrepo_metrics.go

```diff
@@ -0,0 +1,158 @@
+
+	gitrepoResourcesDesiredReady = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: gitRepoSubsystem,
+			Name:      "resources_desired_ready",
+			Help:      "The count of resources that are desired to be in a Ready state.",
+		},
+		gitRepoLabels,
+	)
+	gitrepoResourcesMissing = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: gitRepoSubsystem,
+			Name:      "resources_missing",
+			Help:      "The count of resources that are in a Missing state.",
+		},
+		gitRepoLabels,
+	)
+	gitrepoResourcesModified = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: gitRepoSubsystem,
+			Name:      "resources_modified",
+			Help:      "The count of resources that are in a Modified state.",
+		},
+		gitRepoLabels,
+	)
+	gitrepoResourcesNotReady = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: gitRepoSubsystem,
+			Name:      "resources_not_ready",
+			Help:      "The count of resources that are in a NotReady state.",
+		},
+		gitRepoLabels,
+	)
+	gitrepoResourcesOrphaned = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: gitRepoSubsystem,
+			Name:      "resources_orphaned",
+			Help:      "The count of resources that are in an Orphaned state.",
+		},
+		gitRepoLabels,
+	)
+	gitrepoResourcesReady = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: gitRepoSubsystem,
+			Name:      "resources_ready",
+			Help:      "The count of resources that are in a Ready state.",
+		},
+		gitRepoLabels,
+	)
+	gitrepoResourcesUnknown = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: gitRepoSubsystem,
+			Name:      "resources_unknown",
+			Help:      "The count of resources that are in an Unknown state.",
+		},
+		gitRepoLabels,
+	)
+	gitrepoResourcesWaitApplied = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: gitRepoSubsystem,
+			Name:      "resources_wait_applied",
+			Help:      "The count of resources that are in a WaitApplied state.",
+		},
+		gitRepoLabels,
+	)
+	gitrepoDesiredReadyClusters = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: gitRepoSubsystem,
+			Name:      "desired_ready_clusters",
+			Help:      "The amount of clusters desired to be in a ready state.",
+		},
+		gitRepoLabels,
+	)
+	gitrepoReadyClusters = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: gitRepoSubsystem,
+			Name:      "ready_clusters",
+			Help:      "The count of cluster in a Ready state.",
+		},
+		gitRepoLabels,
+	)
+	gitrepoObserved = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: gitRepoSubsystem,
+			Name:      "observations_total",
+			Help:      "The total times that this GitRepo has been observed",
```

Comment:
I wonder if this phrasing might be a bit ambiguous to users. WDYT? How about something like this?
```suggestion
			Help:      "The total times that metrics have been computed for this GitRepo",
```

Reply:
Because I could not come up with a clear use case and because collecting/updating metrics several times inside a single reconciliation loop is possible, I think keeping it is more confusing than helpful. Therefore, I've removed this metric.